File: leetcode75/go/392-is-subsequence.go
package main

import "fmt"

func main() {
	s := "abc"
	t := "ahbgdc"
	fmt.Println(isSubsequence(s, t))
	s = "axc"
	t = "ahbgdc"
	fmt.Println(isSubsequence(s, t))
}

func isSubsequence(s string, t string) bool {
	sLen, tLen := len(s), len(t)
	if sLen == 0 {
		return true
	}
	sIndex, tIndex := 0, 0
	for tIndex < tLen && sIndex < sLen {
		if s[sIndex] == t[tIndex] {
			sIndex++
		}
		tIndex++
	}
	return sIndex == sLen
}

File: leetcode75/js/392-is-subsequence.js
var isSubsequence = function(s, t) {
    let sIndex = 0;
    let tIndex = 0;
    
    while (sIndex < s.length && tIndex < t.length) {    
        if (s[sIndex] === t[tIndex]) {
            sIndex++;
            tIndex++;
        } else {
            tIndex++;
        }
    }
    return sIndex === s.length;
}

var s = "abc", t = "ahbgdc";
console.log(isSubsequence(s, t)); // true
var s = "axc", t = "ahbgdc";
console.log(isSubsequence(s, t)); // false